clientupdate/clientupdate.go

```diff
@@ -259,13 +269,16 @@ func Update(args Arguments) error {
 }
 
 func (up *Updater) confirm(ver string) bool {
-	switch cmpver.Compare(version.Short(), ver) {
-	case 0:
-		up.Logf("already running %v version %v; no update needed", up.Track, ver)
-		return false
-	case 1:
-		up.Logf("installed %v version %v is newer than the latest available version %v; no update needed", up.Track, version.Short(), ver)
-		return false
+	// Only check version when we're not switching tracks.
+	if up.Track == "" || up.Track == CurrentTrack {
+		switch cmpver.Compare(up.currentVersion, ver) {
```

Comment:
Minor nit, [cmpver](https://pkg.go.dev/tailscale.com/util/cmpver) only promises the values will be non-zero in the right direction, not that they will be exactly 1 or -1

Maybe:

```go
if c := cmpver.Cmpare(up.CurrentVersion, ver); c == 0 {
  // ... log equal
} else if c > 0 {
  // ... log newer
}
```

Reply:
good call, updated!